android/KMEA/app/src/main/java/com/tavultesoft/kmea/KMManager.java

```diff
@@ -405,11 +405,9 @@ public static void onPause() {
   public static void onDestroy() {
     if (InAppKeyboard != null) {
       InAppKeyboard.onDestroy();
-      InAppKeyboard = null;
     }
     if (SystemKeyboard != null) {
       SystemKeyboard.onDestroy();
-      SystemKeyboard = null;
     }
```

Comment:
> Can you explain the code path that would trigger this issue? It bothers me that we continue using those objects after calling `onDestroy`.

My initial test scenarios were with the Keyman app running in the background, Keyman as system keyboard, and then closing the Keyman system keyboard. If it  hadn't crashed, the Keyman app would have a blank keyboard.

Some of the crashes after `onDestroy` occured with the following calls

-----

`KMManager.pageLoaded()` ~ line 1490
```
if (!InAppKeyboard.keyboardSet) {
```
-----

`KMHardwareKeyboardInterpreter.onKeyDown()` ~ line 26
```
boolean isChiral = KMManager.getKMKeyboard(this.keyboardType).getChirality();
```

Reply:
I'm not seeing how the `onKeyDown` event could crash -- unless it's actually in `KMManager.getKMKeyboard`.

So, this looks like a bit of a design issue -- the awkward split between `InAppKeyboard` and `SystemKeyboard`, and then lifecycle is uncertain and not validated in `KMManager`. A better long-term fix would be to ensure that we had valid state at any entry point in `KMManager` (e.g. `if(InAppKeyboard != null) ...`.

But that may be too much work for today; if so, feel free to turn this into an issue for the future.

Reply:
k, that will be deferred in #2543